test/Microsoft.Sbom.Tool.Tests/IntegrationTests.cs

```diff
@@ -138,6 +139,40 @@ public void E2E_GenerateAndValidateSPDX30Manifest_ValidationSucceeds_ReturnsZero
         Assert.IsTrue(File.ReadAllText(outputFile).Contains("\"Result\":\"Success\"", StringComparison.OrdinalIgnoreCase));
     }
 
+    [TestMethod]
+    public void E2E_CompareSPDX22AndSPDX30Manifests_ComparisonAndDeterminismSucceeds()
+    {
+        if (!IsWindows)
+        {
+            Assert.Inconclusive("This test is not (yet) supported on non-Windows platforms.");
+            return;
+        }
+
+        // Generate SPDX 2.2 manifest
+        var spdx22TestFolderPath = CreateTestFolder();
+        GenerateManifestAndValidateSuccess(spdx22TestFolderPath);
+        var spdx22ManifestPath = Path.Combine(AppendFullManifestFolderPath(spdx22TestFolderPath), ManifestFileName);
+        var spdx22Json = ReadJsonFile(spdx22ManifestPath);
+
+        if (Directory.Exists(spdx22TestFolderPath))
+        {
+            Directory.Delete(spdx22TestFolderPath, true);
+        }
+
+        // Generate SPDX 3.0 manifest
+        var spdx30TestFolderPath = CreateTestFolder();
+        GenerateManifestAndValidateSuccess(spdx30TestFolderPath, manifestInfoSpdxVersion: "3.0");
+        var spdx30ManifestPath = Path.Combine(AppendFullManifestFolderPath(spdx30TestFolderPath, "3.0"), ManifestFileName);
+        var spdx30Json = ReadJsonFile(spdx30ManifestPath);
+
+        // Use SbomEqualityComparer to compare the two manifests
+        var comparer = new SbomEqualityComparer(spdx22Json, spdx30Json);
+        var areEqual = comparer.DocumentsEqual();
+
+        // Assert that the manifests are equal
+        Assert.IsTrue(areEqual, "The SPDX 2.2 and SPDX 3.0 manifests should be equivalent.");
```

Comment:
Would it be interesting/possible to have unit tests where we tweaked one aspect at a time to ensure that we pin how they differ?

Reply:
There's lots of different combinations of how the docs can differ, are there any specific cases you're thinking of that we should target?